common/cpp/react/renderer/components/rnscreens/RNSScreenComponentDescriptor.h

```diff
@@ -26,13 +41,103 @@ class RNSScreenComponentDescriptor final
             shadowNode.getState());
     auto stateData = state->getData();
 
+#ifdef ANDROID
     if (stateData.frameSize.width != 0 && stateData.frameSize.height != 0) {
       layoutableShadowNode.setSize(
```

Comment:
Could we have some comparison if the pre-calculated values were same as the ones in `stateData` and e.g. log it somehow if they weren't? That is a very valuable information that we are calculating something wrong and it needs a fix ASAP.

Reply:
I mean, we could compare header height received from the native side & see whether it matches the value we computed in the very first layout - however right now we pass invalid value there (as `contentOffsetY`) and it requires separate fix that I believe is outside the scope of this PR. 

I can add a check inside this `if`, that checks whether the very first frame received from UI is the same as the frame computed during the very first layout. That's ok?

Reply:
Yeah, that's what I meant. If they are the same, then the views should physically be in the same place yeah?

Reply:
Done in: https://github.com/software-mansion/react-native-screens/pull/2169/commits/e3c2fd89f531562c72dd5d62c150b5a28de90380